compiler/src/codegen/comp_utils.re

```diff
@@ -195,152 +165,147 @@ let type_of_repr = repr => {
-                  func_ptr,
-                  arguments,
-                  call_arg_types,
-                  call_result_types,
-                );
-              | Unknown => failwith("Impossible: Unknown function call type")
-              };
-            let function_body =
-              switch (rets) {
-              | [] => Expression.Drop.make(wasm_mod, function_call)
-              | _ => function_call
-              };
-            let function_body =
-              Expression.Block.make(
-                wasm_mod,
-                "closure_incref",
-                [
-                  Expression.If.make(
+                    Expression.Const.make(wasm_mod, Literal.int32(8l)),
+                  ),
+                  Expression.Binary.make(
                     wasm_mod,
-                    Expression.Binary.make(
-                      wasm_mod,
-                      Op.ne_int32,
-                      get_closure(),
-                      Expression.Const.make(wasm_mod, Literal.int32(0l)),
-                    ),
-                    store(
+                    Op.add_int32,
+                    load(
                       wasm_mod,
                       Expression.Binary.make(
                         wasm_mod,
                         Op.sub_int32,
                         get_closure(),
                         Expression.Const.make(wasm_mod, Literal.int32(8l)),
                       ),
-                      Expression.Binary.make(
-                        wasm_mod,
-                        Op.add_int32,
-                        load(
-                          wasm_mod,
-                          Expression.Binary.make(
-                            wasm_mod,
-                            Op.sub_int32,
-                            get_closure(),
-                            Expression.Const.make(
-                              wasm_mod,
-                              Literal.int32(8l),
-                            ),
-                          ),
-                        ),
-                        Expression.Const.make(wasm_mod, Literal.int32(1l)),
-                      ),
                     ),
-                    Expression.Null.make(),
+                    Expression.Const.make(wasm_mod, Literal.int32(1l)),
                   ),
-                  function_body,
-                ],
-              );
-            let arg_types =
-              Type.create(Array.of_list(List.map(type_of_repr, args)));
-            let result_types =
-              Type.create(Array.of_list(List.map(type_of_repr, rets)));
-            ignore @@
-            Function.add_function(
-              wasm_mod,
-              name,
-              arg_types,
-              result_types,
-              [||],
+                ),
+                Expression.Null.make(),
+              ),
               function_body,
-            );
-            // Remove existing Grain export (if any)
-            Export.remove_export(wasm_mod, name);
```

Comment:
Why is the remove_export no longer necessary

Reply:
When we used wasm files as object files, we wrote special Grain export names in the Wasm binaries to facilitate linking later. We don't do this anymore before linking happens before we generate wasm (so there was no Grain export in the first place to worry about removing).